source/transactions/tests/mongos-recovery-token.yml

```diff
@@ -0,0 +1,362 @@
+database_name: &database_name "transaction-tests"
+collection_name: &collection_name "test"
+
+data: []
+
+tests:
+  - description: commitTransaction explicit retries succeed on new mongos
+    topology: ["sharded"]
```

Comment:
It looks like this entire test pertains to mongos, so that might agree with making `topology` a top-level field.

Reply:
Let's continue this conversation on the other comment thread: https://github.com/mongodb/specifications/pull/471/files/1562085a8001c04464a0efaf5fd134d6cf7c97df#r255563911